dashboard/public/frequency/frequency.js

```diff
@@ -0,0 +1,879 @@
+
+var ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+var LETTERS = ALPHABET.split('');
+var ENGLISH = {
+  A: 0.08167,
+  B: 0.01492,
+  C: 0.02782,
+  D: 0.04253,
+  E: 0.12702,
+  F: 0.02228,
+  G: 0.02015,
+  H: 0.06094,
+  I: 0.06966,
+  J: 0.00153,
+  K: 0.00772,
+  L: 0.04025,
+  M: 0.02406,
+  N: 0.06749,
+  O: 0.07507,
+  P: 0.01929,
+  Q: 0.00095,
+  R: 0.05987,
+  S: 0.06327,
+  T: 0.09056,
+  U: 0.02758,
+  V: 0.00978,
+  W: 0.02361,
+  X: 0.0015,
+  Y: 0.01974,
+  Z: 0.00074
+};
+
+// Debounce function, courtesy of
+// http://davidwalsh.name/function-debounce
+// Currently unused.
+var debounce = function (func, wait, immediate) {
+  var timeout;
+  return function () {
+    var context = this,
+      args = arguments;
+    var later = function () {
+      timeout = null;
+      if (!immediate) func.apply(context, args);
+    };
+    var callNow = immediate && !timeout;
+    clearTimeout(timeout);
+    timeout = setTimeout(later, wait);
+    if (callNow) func.apply(context, args);
+  };
+};
+
+/**
+ * @typedef {Object} UserData
+ * @property {string} letter - the single uppercase alphabetical
+ *                    character this data point represents
+ * @property {number} frequency - a value between 0 and 1 representing
+ *                    the relative frequency of this character in the
+ *                    user input.
+ * @property {boolean} locked - whether or not this value has been
+ *                     bound (semi-) permanently to a corresponding
+ *                     english letter
+ */
+
+/**
+ * @typedef {Object} EnglishData
+ * @property {string} letter - the single uppercase alphabetical
+ *                    character this data point represents
+ * @property {number} frequency - a value between 0 and 1 representing
+ *                    the relative frequency of this character in
+ *                    standard english
+ */
+
+/**
+ * Master class for rendering the bar graph, including all interactive
+ * components
+ *
+ * @param {!Object} options
+ * @param {!JQuery} options.chart_container - the DOM element into which
```

Comment:
Nit: The correct capitalization for this type (contrary to our style guide) is [`jQuery`](http://api.jquery.com/Types/#jQuery).  I only notice because RubyMine's linter warns me about it.